pkgs/tools/networking/curl/default.nix

```diff
@@ -76,8 +76,8 @@ stdenv.mkDerivation rec {
     ++ stdenv.lib.optional c-aresSupport "--enable-ares=${c-ares}"
     ++ stdenv.lib.optional gssSupport "--with-gssapi=${kerberos.dev}";
 
-  CXX = "c++";
-  CXXCPP = "c++ -E";
+  CXX = "${stdenv.cc.targetPrefix}c++";
+  CXXCPP = "${stdenv.cc.targetPrefix}c++ -E";
```

Comment:
This works as-is for me as well?

```console
$ nix build -f channel:nixos-unstable --arg crossSystem '(import <nixpkgs/lib>).systems.examples.aarch64-multiplatform-musl' curl
```

Reply:
I'll admit I didn't test this one. I had assumed that `c++` would either not exist or refer to the host compiler.